### PR TITLE
Coverage support via coveralls

### PR DESCRIPTION
**Pull request by captn3m0**
- Adds rubocop as a style check
- Upgrades the dependencies
- Fixes style issues raised by rubocop

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -2,6 +2,7 @@ require 'rake/testtask'
 
 Rake::TestTask.new do |t|
   t.libs << 'test'
+  t.warning = false
```
Comment (harman28) [inline Rakefile]: I don't think this is necessary. Only warnings are circular dependency warnings, which I'm getting rid of anyway, and one from webmock, which should exist if only to remind us to upgrade.

Comment (captn3m0) [inline Rakefile]: If we merge this first, you can flip it to warning = true in v2 branch?

Comment (harman28) [inline Rakefile]: Makes sense.

File: test/test_helper.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
-require 'simplecov'
+require 'coveralls'
```
Comment (harman28) [inline test/test_helper.rb]: Won't `coveralls` have to be added as a development dependency in the spec as well?

Comment (captn3m0) [inline test/test_helper.rb]: It is added

**Pull request merged**